virtool/tasks/runner.py

```diff
@@ -12,45 +11,90 @@
 from sentry_sdk import capture_exception
 
 
+async def shutdown_task_runner(app: Application):
+    """
+    Raise an `asyncio.CancelledError` in the running task runner.
+
+    :param app: Current running application.
+    """
+    await app["task_runner"].close(timeout=float("inf"))
```

Comment:
1. Calls to `close()` shouldn't be necessary as the entire `aiojobs` scheduler is closed on application shutdown.

   Is this not the case?

2. All (necessary) shutdown functions need to go in `virtool.shutdown`.

Reply:
Directly using `scheduler.close()` was resulting in multiple `asycio.cancelledError`s being propagated in short succession. I think this can be fixed by increasing the timeout from the default `0.1s` to a larger value. This will make it apply to all jobs, but that's probably not a huge deal. Inside the component then it leaves the options of either assuming the second `cancelledError` is the result of the timeout or putting a second timeout inside the component that is slightly shorter to reduce the chances of a race condition. I wasn't a fan of either option, so suggested doing a separate shutdown. I am not stuck on either soln, but thought I would throw in some extra context

Reply:
Do you know why more than one `CancelledError` is raised inside the runner?

Looking at the source for `aiojobs`, I don't see how a subsequenty `CancelledError` would be raised. If the timeout is reached, a `TimeoutError` should be raised.

https://github.com/aio-libs/aiojobs/blob/522d10f2de04ee2cc3088fd962047847d40efa97/aiojobs/_job.py#LL117C9-L135C54

Reply:
@markus Please resolve the other comments here. Reece will deal with this when you are done.